### DISCUSSION TITLE
Не получается настроить работу по hostlist

### DISCUSSION THREAD

**Discussion by AloofLibra** [bol-van/zapret#910] [Q&A] [answered]
Пытаюсь настроить работу через hostlist, но почему-то стратегия бьёт и по сайтам, которые не добавлены ни в один список. 
При этом если добавить такие сайты в exclude то всё прекрасно работает... Но добавлять половину интернета в exclude так себе решение. 
Стратегия 
`--filter-tcp=443 --dpi-desync=syndata,disorder2 --dpi-desync-fake-syndata=/opt/zapret/files/fake/tls_clienthello_iana_org.bin <HOSTLIST> --new`
MODE_FILTER=autohostlist
OS: `OpenWrt SNAPSHOT, r27671`

Искренне не понимаю что я делаю не так... В листе автохостлист пусто. 

Заранее спасибо

**Comment by numbereleven-a** [comment]
а кинь полный конфиг от NFQWS_OPT=" до "

**Reply by AloofLibra** [reply]
NFQWS_OPT="
--filter-tcp=80 --dpi-desync=syndata,disorder2 --dpi-desync-fake-syndata=/opt/zapret/files/fake/tls_clienthello_iana_org.bin <HOSTLIST> --new
--filter-tcp=443 --dpi-desync=syndata,disorder2 --dpi-desync-fake-syndata=/opt/zapret/files/fake/tls_clienthello_iana_org.bin <HOSTLIST> --new
--filter-udp=443 --dpi-desync=fake --dpi-desync-repeats=3 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin <HOSTLIST>
"

**Reply by numbereleven-a** [reply]
попробуй указать HOSTLIST, не понятно какие сайты должны дуриться, ну и в zapret-hosts-user.txt добавь сайты которые надо обходить
PS
Поиск по "Параметры nfqws" там будет пример
https://github.com/bol-van/zapret?tab=readme-ov-file#%D0%B1%D1%8B%D1%81%D1%82%D1%80%D1%8B%D0%B9-%D1%81%D1%82%D0%B0%D1%80%D1%82

**Reply by AloofLibra** [reply]
А, прикольно, гитхаб вырезал мне < HOSTLIST> в конце почему-то

**Reply by AloofLibra** [reply]
При запуске хостлисты подтягиваются судя по выводу

`Starting daemon 3: /opt/zapret/nfq/nfqws --qnum=200 --user=daemon --dpi-desync-fwmark=0x40000000

--filter-tcp=80 --dpi-desync=syndata,disorder2 --dpi-desync-fake-syndata=/opt/zapret/files/fake/tls_clienthello_iana_org.bin --hostlist=/opt/zapret/ipset/zapret-hosts-user.txt --hostlist-exclude=/opt/zapret/ipset/zapret-hosts-user-exclude.txt --hostlist-auto=/opt/zapret/ipset/zapret-hosts-auto.txt --hostlist-auto-fail-threshold=3 --hostlist-auto-fail-time=60 --hostlist-auto-retrans-threshold=3 --new

--filter-tcp=443 --debug=1 --dpi-desync=syndata,disorder2 --dpi-desync-fake-syndata=/opt/zapret/files/fake/tls_clienthello_iana_org.bin --hostlist=/opt/zapret/ipset/zapret-hosts-user.txt --hostlist-exclude=/opt/zapret/ipset/zapret-hosts-user-exclude.txt --hostlist-auto=/opt/zapret/ipset/zapret-hosts-auto.txt --hostlist-auto-fail-threshold=3 --hostlist-auto-fail-time=60 --hostlist-auto-retrans-threshold=3 --new

--filter-udp=443 --dpi-desync=fake --dpi-desync-repeats=3 --dpi-desync-fake-quic=/opt/zapret/files/fake/quic_initial_www_google_com.bin --hostlist=/opt/zapret/ipset/zapret-hosts-user.txt --hostlist-exclude=/opt/zapret/ipset/zapret-hosts-user-exclude.txt --hostlist-auto=/opt/zapret/ipset/zapret-hosts-auto.txt --hostlist-auto-fail-threshold=3 --hostlist-auto-fail-time=60 --hostlist-auto-retrans-threshold=3`

**Reply by numbereleven-a** [reply, marked as answer]
не обратил внимание что syndata
syndata это нулевая фаза, с ней листы не работают, поищи другую стратегию
PS
КОМБИНИРОВАНИЕ МЕТОДОВ ДЕСИНХРОНИЗАЦИИ

В параметре dpi-desync можно указать до 3 режимов через запятую.

    0 фаза - предполагает работу на этапе установления соединения : synack, syndata, --wsize, --wssize. На эту фазу не действуют фильтры по [hostlist](https://github.com/bol-van/zapret?tab=readme-ov-file#%D0%BC%D0%BD%D0%BE%D0%B6%D0%B5%D1%81%D1%82%D0%B2%D0%B5%D0%BD%D0%BD%D1%8B%D0%B5-%D1%81%D1%82%D1%80%D0%B0%D1%82%D0%B5%D0%B3%D0%B8%D0%B8).
    1 фаза - отсылка чего-либо до оригинального пакета данных : fake, rst, rstack.
    2 фаза - отсылка в модифицированном виде оригинального пакета данных (например, fakedsplit или ipfrag2).

**Reply by AloofLibra** [reply]
О как... Спасибо, буду знать) А то весь мозг сломал уже... А эта часть как-то мимо меня проехала

**Reply by OverLocker** [reply]
Подскажите, пожалуйста. У меня получается что по умолчанию сервис работает как фейк вообще для всего и надо вносить исключения. А можно ли только для нужного, а остальное пропускать напрямую, и еще желательно отрезолвить адреса (вроде раньше так было на какой-то версии)? Спасибо

**Reply by numbereleven-a** [reply]
> Подскажите, пожалуйста. У меня получается что по умолчанию сервис работает как фейк вообще для всего и надо вносить исключения. А можно ли только для нужного, а остальное пропускать напрямую, и еще желательно отрезолвить адреса (вроде раньше так было на какой-то версии)? Спасибо

Добавить сайты которые должны обходиться в zapret-hosts-user.txt, а которые не должны обходиться, в zapret-hosts-user-exclude.txt, поддомены учитываются автоматически, в стратегиях указать параметр `<HOSTLIST>`
Способ работает для режимов фильтрации MODE_FILTER=hostlist или MODE_FILTER=autohostlist.

отрезолвить
/opt/zapret/ipset/./clear_lists.sh
/opt/zapret/ipset/./get_config.sh

**Reply by OverLocker** [reply]
Идеально. Все сработало так, как и хотелось. Большое спасибо.
